core/trino-main/src/main/java/io/trino/sql/gen/PageFunctionCompiler.java

```diff
@@ -203,11 +204,12 @@ private Supplier<PageProjection> compileProjectionInternal(RowExpression project
             throw new TrinoException(COMPILER_ERROR, e);
         }
 
+        MethodHandle pageProjectionConstructor = constructorMethodHandle(pageProjectionWorkClass, BlockBuilder.class, ConnectorSession.class, Page.class, SelectedPositions.class);
         return () -> new GeneratedPageProjection(
                 result.getRewrittenExpression(),
                 isDeterministic(result.getRewrittenExpression()),
```

Comment:
We could also pull out `isDeterministic`. I don't think that needs to be checked repeatedly.